lib/postgres_ext/active_record/relation/predicate_builder.rb

```diff
@@ -7,7 +7,7 @@ def self.build(attribute, value)
       when Array
         engine = attribute.relation.engine
         column = engine.connection.schema_cache.columns(attribute.relation.name).detect{ |col| col.name.to_s == attribute.name.to_s }
-        if column && column.array
+        if column && column.respond_to?(:array) && column.array
```

Comment:
This pattern is already in quite a few places, I was just sticking with convention - you're right though.